crates/demo/src/main.rs

```diff
@@ -24,36 +24,40 @@ async fn run(event_loop: EventLoop<()>, window: Window) {
     let mut graphics = Graphics::new(&window, 4).await;
 
     event_loop.set_control_flow(ControlFlow::Poll);
+    // FIXME update this as well
     event_loop
-        .run(move |event, elwt| {
-            if graphics.handle_event(&mut yak, &event, elwt) {
-                return;
+        .run(move |event, event_loop| match event {
+            Event::AboutToWait => {
+                window.request_redraw();
             }
 
-            match event {
-                Event::AboutToWait => {
-                    window.request_redraw();
-                }
+            Event::WindowEvent {
+                event: WindowEvent::RedrawRequested { .. },
+                ..
+            } => {
+                yak.start();
+                app();
+                yak.finish();
 
-                Event::WindowEvent {
-                    event: WindowEvent::RedrawRequested { .. },
-                    ..
-                } => {
-                    yak.start();
-                    app();
-                    yak.finish();
+                graphics.paint(&mut yak, wgpu::Color::BLACK);
+            }
 
-                    graphics.paint(&mut yak, wgpu::Color::BLACK);
+            Event::WindowEvent { event, .. } => {
+                if graphics.handle_event(&mut yak, &event, event_loop) {
+                    return;
```

Comment:
Can we preserve the original structure of the code here? The hope is that `handle_event` runs _before_ your match block and helps filter out events. It should be able to accept the `Event` type just like before.

Reply:
see the comment on DeviceEvent